config/initializers/content_security_policy.rb

```diff
@@ -45,7 +51,12 @@
 end
 
 # Generate session nonces for permitted importmap, inline scripts, and inline styles.
-Rails.application.config.content_security_policy_nonce_generator = ->(request) { request.session.id.to_s }
+Rails.application.config.content_security_policy_nonce_generator = lambda { |request|
+  # Suggested nonce generator doesn't work on first page load https://github.com/rails/rails/issues/48463
+  # Related PR attempting to fix: https://github.com/rails/rails/pull/48510
+  request.session.send(:load_for_write!) # force session to be created
+  request.session.id.to_s.presence || raise("No session ID available in #{request.inspect}")
```

Comment:
remove the raise?

Reply:
Replaced with a fallback to SecureRandom.base64(16)